src/module.c

```diff
@@ -8041,8 +8023,11 @@ int RM_SubscribeToKeyspaceEvents(RedisModuleCtx *ctx, int types, RedisModuleNoti
 }
 
 void firePostExecutionUnitJobs() {
-    /* Avoid propagation of commands. */
-    server.in_nested_call++;
+    /* Avoid propagation of commands.
+     * In that way, postExecutionUnitOperations will prevent
+     * recursive calls to firePostExecutionUnitJobs. */
+     // TODO: All command propagated from within this function will be wrapped in MULTI/EXEC - is this what we want?
```

Comment:
Yes, we want all them all to be atomic.

Reply:
but doesn't it mean that propagation from different modules will be in the same transaction together?

Reply:
It is, and this is what we want. Post notification job API promise that the job will be called atomically when it is safe to write.

Reply:
ok, but maybe we want to wrap each module's commands in MULTI/EXEC, instead of one MULTI/EXEC that contains commands from several modules..? idk it feels a bit weird

Reply:
I do not think so, what if one modules performs some write that trigger notification on another module. You want all the effect to be atomic (wrap with multi exec).